extension/src/pipeline/quickPick.ts

```diff
@@ -79,36 +95,79 @@ const getFieldOptions = (data: UnknownValue): string[] => {
     : []
 }
 
-export const pickPlotConfiguration = async (): Promise<
-  PlotConfigData | undefined
-> => {
-  const file = await pickDataFile()
+const showNotEnoughKeysToast = () =>
+  Toast.showError(
+    'The requested file(s) does not contain enough keys (columns) to generate a plot. Does the file or files follow the DVC plot guidelines for [JSON/YAML](https://dvc.org/doc/command-reference/plots/show#example-hierarchical-data) or [CSV/TSV](https://dvc.org/doc/command-reference/plots/show#example-tabular-data) files?'
+  )
+
+const getFieldsFromDataFiles = (
+  dataArr: { data: UnknownValue; file: string }[]
+) => {
+  const keys: {
+    [file: string]: string[]
+  } = {}
+  let keysAmount = 0
+
+  for (const { file, data } of dataArr) {
+    const fields = getFieldsFromValue(data)
+
+    if (fields.length === 0) {
```

Comment:
[Q] Should we do this check after we've collected all of the keys? If not I think we need to specify which of the files was the one that caused the process to reject.

Reply:
> [Q] Should we do this check after we've collected all of the keys? If not I think we need to specify which of the files was the one that caused the process to reject.

Makes sense! I can adjust the toast message to mention which file is failing.